sql/log_event.h

```diff
@@ -3666,7 +3666,19 @@ class Gtid_log_event: public Log_event
   {
     return GTID_HEADER_LEN + ((flags2 & FL_GROUP_COMMIT_ID) ? 2 : 0);
   }
-  bool is_valid() const { return seq_no != 0; }
+
+  bool is_valid() const
+  {
+    /*
+      If the GTID event indicates that it will use extra engines, there must be
+      extra engines set
+    */
+    if (flags_extra && (flags_extra & FL_EXTRA_MULTI_ENGINE) && !extra_engines)
+      return false;
+
+    return seq_no != 0;
+  }
```

Comment:
The other existing check for length (FL_GROUP_COMMIT_ID) sets seq_no=0 to mark that the event is not valid. Why not do the same for the new check you added? For consistency, and to pay the extra price only in the rare error case, not in the common case of checking is_valid() for correct event?

Reply:
I missed that the earlier code sets `seq_no=0` to invalidate the event. Agreed that it would be better to have the consistency.